packages/gitbook/src/components/SiteSectionTabs/SectionIcon.tsx

```diff
@@ -0,0 +1,15 @@
+import type { SiteSection } from '@gitbook/api';
+import { Icon, type IconName } from '@gitbook/icons';
+
+import { type ClassValue, tcls } from '@/lib/tailwind';
+
+/**
+ * Icon shown beside a section in the site section tabs.
+ */
+export function SectionIcon(props: { section: SiteSection; className?: ClassValue }) {
+    const { section, className } = props;
+
+    return section.icon ? (
+        <Icon icon={section.icon as IconName} className={tcls('size-[1em]', className)} />
```

Comment:
I feel we don't need to check it again if we do assertion already in the parent. This component should maybe only accept `icon` prop of the section